fdbclient/include/fdbclient/TenantManagement.actor.h

```diff
@@ -257,6 +275,17 @@ Future<Optional<TenantMapEntry>> createTenant(Reference<DB> db,
 				checkExistence = false;
 			}
 
+			state Future<int64_t> tenantIdFuture;
+			if (generateTenantId) {
+				Optional<int64_t> existingId = wait(TenantMetadata::tenantNameIndex().get(tr, name));
+				// If a tenant already exists then the id here doesnt matter since the tenant will not be created
+				if (existingId.present()) {
+					tenantIdFuture = existingId.get();
+				} else {
+					tenantIdFuture = getNextTenantId(tr, 1);
+				}
+			}
```

Comment:
Why was this block was moved down? It's now running less concurrently and is fetching the tenant name index twice, so we should see if we can meet our new needs while keeping the old properties.

Reply:
I'll change it so we dont fetch the tenant name index twice